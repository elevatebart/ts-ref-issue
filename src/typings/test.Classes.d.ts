import { Theme } from './RsgTheme';

declare global {
	/**
	 * function used in react tests to generate
	 * mocks of JSS Class names
	 */
	const classes: (styles: (theme: Theme) => Record<string, any>) => Record<string, string>;
}
