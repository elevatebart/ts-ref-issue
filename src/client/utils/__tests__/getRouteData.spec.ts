import deepfreeze from 'deepfreeze';
import getRouteData from '../getRouteData';
import { DisplayModes } from '../../consts';
import * as Rsg from '../../../typings';

const example0: Rsg.Example = {
	type: 'code',
	content: 'const a = 0',
	evalInContext: () => () => 3,
};

const example1: Rsg.Example = {
	type: 'markdown',
	content: '# title',
};

const sections: Rsg.Section[] = deepfreeze([
	{
		sections: [
			{
				name: 'Components',
				slug: 'components',
				components: [
					{
						name: 'Button',
						props: {
							displayName: 'Button',
							examples: [example0, example1],
						},
						module: 1,
					},
					{
						name: 'Image',
						props: {
							displayName: 'Image',
						},
						module: 2,
					},
				],
				sections: [],
				exampleMode: 'collapse',
				usageMode: 'collapse',
				sectionDepth: 0,
			},
			{
				name: 'Section',
				slug: 'section',
				content: [example0, example1],
				components: [],
				sections: [],
				exampleMode: 'collapse',
				usageMode: 'collapse',
				sectionDepth: 0,
			},
			{
				name: 'Buttons',
				slug: 'buttons',
				components: [
					{
						name: 'Label',
						module: 1,
					},
					{
						name: 'RandomButton',
						module: 2,
					},
				],
				sections: [],
				exampleMode: 'collapse',
				usageMode: 'collapse',
				sectionDepth: 2,
			},
		],
	},
]);

describe('getRouteData', () => {
	it('should return "all" mode by default', () => {
		const result = getRouteData([], '');
		expect(result.displayMode).toBe(DisplayModes.all);
	});

	it('should return one component', () => {
		const result = getRouteData(sections, '#!/Button');
		expect(result).toMatchSnapshot();
	});

	it('should return one section', () => {
		const result = getRouteData(sections, '#!/Section');
		expect(result).toMatchSnapshot();
	});

	it('should return one example from a component', () => {
		const result = getRouteData(sections, '#!/Button/1');
		expect(result).toMatchSnapshot();
	});

	it('should return one example from a section', () => {
		const result = getRouteData(sections, '#!/Section/1');
		expect(result).toMatchSnapshot();
	});

	it('should return first section if pagePerSection and hash is empty', () => {
		const subSection = sections[0].sections as Rsg.Section[];
		const result = getRouteData(subSection, '', true);
		expect(result).toMatchSnapshot();
	});

	it('should return one section if pagePerSection and hash is #/Section', () => {
		const result = getRouteData(sections, '#/Section', true);
		expect(result).toMatchSnapshot();
	});

	it('should return one section without components if pagePerSection and hash is #/Buttons', () => {
		const result = getRouteData(sections, '#/Buttons', true);
		expect(result).toMatchSnapshot();
	});

	it('should return one component if pagePerSection and hash is #/Buttons/Label', () => {
		const result = getRouteData(sections, '#/Buttons/Label', true);
		expect(result).toMatchSnapshot();
	});

	it('should return not found if pagePerSection and hash is #/Buttons/Label/Not', () => {
		const result = getRouteData(sections, '#/Buttons/Label/Not', true);
		expect(result).toMatchSnapshot();
	});
});
