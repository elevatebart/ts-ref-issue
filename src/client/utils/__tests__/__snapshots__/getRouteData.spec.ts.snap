// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRouteData should return first section if pagePerSection and hash is empty 1`] = `
Object {
  "displayMode": "all",
  "sections": Array [
    Object {
      "components": Array [
        Object {
          "module": 1,
          "name": "Button",
          "props": Object {
            "displayName": "Button",
            "examples": Array [
              Object {
                "content": "const a = 0",
                "evalInContext": [Function],
                "type": "code",
              },
              Object {
                "content": "# title",
                "type": "markdown",
              },
            ],
          },
        },
        Object {
          "module": 2,
          "name": "Image",
          "props": Object {
            "displayName": "Image",
          },
        },
      ],
      "exampleMode": "collapse",
      "name": "Components",
      "sectionDepth": 0,
      "sections": Array [],
      "slug": "components",
      "usageMode": "collapse",
    },
  ],
}
`;

exports[`getRouteData should return not found if pagePerSection and hash is #/Buttons/Label/Not 1`] = `
Object {
  "displayMode": "notFound",
  "sections": Array [],
}
`;

exports[`getRouteData should return one component 1`] = `
Object {
  "displayMode": "component",
  "sections": Array [
    Object {
      "components": Array [
        Object {
          "module": 1,
          "name": "Button",
          "props": Object {
            "displayName": "Button",
            "examples": Array [
              Object {
                "content": "const a = 0",
                "evalInContext": [Function],
                "type": "code",
              },
              Object {
                "content": "# title",
                "type": "markdown",
              },
            ],
          },
        },
      ],
      "exampleMode": "collapse",
      "slug": "components",
      "usageMode": "collapse",
    },
  ],
}
`;

exports[`getRouteData should return one component if pagePerSection and hash is #/Buttons/Label 1`] = `
Object {
  "displayMode": "all",
  "sections": Array [
    Object {
      "components": Array [
        Object {
          "module": 1,
          "name": "Label",
        },
      ],
      "exampleMode": "collapse",
      "slug": "buttons",
      "usageMode": "collapse",
    },
  ],
}
`;

exports[`getRouteData should return one example from a component 1`] = `
Object {
  "displayMode": "example",
  "sections": Array [
    Object {
      "components": Array [
        Object {
          "module": 1,
          "name": "Button",
          "props": Object {
            "displayName": "Button",
            "examples": Array [
              Object {
                "content": "# title",
                "type": "markdown",
              },
            ],
          },
        },
      ],
      "exampleMode": "collapse",
      "slug": "components",
      "usageMode": "collapse",
    },
  ],
}
`;

exports[`getRouteData should return one example from a section 1`] = `
Object {
  "displayMode": "example",
  "sections": Array [
    Object {
      "components": Array [],
      "content": Array [
        Object {
          "content": "# title",
          "type": "markdown",
        },
      ],
      "exampleMode": "collapse",
      "name": "Section",
      "sectionDepth": 0,
      "sections": Array [],
      "slug": "section",
      "usageMode": "collapse",
    },
  ],
}
`;

exports[`getRouteData should return one section 1`] = `
Object {
  "displayMode": "section",
  "sections": Array [
    Object {
      "components": Array [],
      "content": Array [
        Object {
          "content": "const a = 0",
          "evalInContext": [Function],
          "type": "code",
        },
        Object {
          "content": "# title",
          "type": "markdown",
        },
      ],
      "exampleMode": "collapse",
      "name": "Section",
      "sectionDepth": 0,
      "sections": Array [],
      "slug": "section",
      "usageMode": "collapse",
    },
  ],
}
`;

exports[`getRouteData should return one section if pagePerSection and hash is #/Section 1`] = `
Object {
  "displayMode": "all",
  "sections": Array [
    Object {
      "components": Array [],
      "content": Array [
        Object {
          "content": "const a = 0",
          "evalInContext": [Function],
          "type": "code",
        },
        Object {
          "content": "# title",
          "type": "markdown",
        },
      ],
      "exampleMode": "collapse",
      "name": "Section",
      "sectionDepth": 0,
      "sections": Array [],
      "slug": "section",
      "usageMode": "collapse",
    },
  ],
}
`;

exports[`getRouteData should return one section without components if pagePerSection and hash is #/Buttons 1`] = `
Object {
  "displayMode": "all",
  "sections": Array [
    Object {
      "components": Array [],
      "exampleMode": "collapse",
      "name": "Buttons",
      "sectionDepth": 2,
      "sections": Array [],
      "slug": "buttons",
      "usageMode": "collapse",
    },
  ],
}
`;
